contracts/base/reward/Bribe.sol

```diff
@@ -59,8 +71,39 @@ contract Bribe is IBribe, MultiRewardsPoolBase {
 
   /// @dev Used to notify a gauge/bribe of a given reward,
   ///      this can create griefing attacks by extending rewards
-  function notifyRewardAmount(address token, uint amount) external override {
-    _notifyRewardAmount(token, amount);
+  function notifyRewardAmount(address token, uint amount) external lock override {
+    _notifyRewardAmount(token, amount, true);
+  }
+
+  /// @dev Add delayed rewards for the next epoch
+  function notifyForNextEpoch(address token, uint amount) external lock override {
+    require(isRewardToken[token], "Token not allowed");
+
+    IERC20(token).safeTransferFrom(msg.sender, address(this), amount);
+
+    uint _epoch = epoch() + 1;
+    rewardsQueue[token][_epoch] = amount;
+
+    emit RewardsForNextEpoch(token, _epoch, amount);
+  }
+
+  /// @dev Notify delayed rewards
+  function notifyDelayedRewards(address token, uint _epoch) external lock override {
+    require(epoch() == _epoch, "!epoch");
+    _notifyDelayedRewards(token, _epoch);
+  }
+
+  function _notifyDelayedRewards(address token, uint _epoch) internal {
+    uint amount = rewardsQueue[token][_epoch];
+    if (amount != 0 && amount > left(token)) {
+      _notifyRewardAmount(token, amount, false);
+      delete rewardsQueue[token][_epoch];
+      emit DelayedRewardsNotified(token, epoch(), amount);
+    }
+  }
+
+  function epoch() public view returns(uint) {
```

Comment:
no, need to read from minter
it suppose to be this number, but can be frontrunned - new epoch in minter can start later
somebody can notify rewards, then change votes and start the epoch